Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -875,20 +928,20 @@ static String mutationObserverNotificationString()
     if (m_size.isEmpty() || documentSize().isEmpty())
         return;
 
-    bool shouldPaintSelection = behavior == PaintingBehavior::All;
-
     auto stateSaver = GraphicsContextStateSaver(context);
 
     auto showDebugIndicators = shouldShowDebugIndicators();
 
     bool haveSelection = false;
     bool isVisibleAndActive = false;
-    if (m_currentSelection && shouldPaintSelection) {
-        // FIXME: Also test is m_currentSelection is not empty.
+#if !ENABLE(UNIFIED_PDF_PAINT_SELECTIONS)
+    bool shouldPaintSelection = behavior == PaintingBehavior::All;
```

Comment:
shouldn't `shouldPaintSelection` be the one that takes UNIFIED_PDF_PAINT_SELECTIONS into account?

Reply:
Indeed, I've changed this logic a little bit and used the helper `canPaintSelectionIntoOwnedLayer()` in https://github.com/WebKit/WebKit/commit/ab434ecd8a237eb5256d62fbc8744be2db5813fc